relay-server/src/services/processor.rs

```diff
@@ -1852,7 +1859,11 @@ impl EnvelopeProcessorService {
         for bucket in &mut buckets {
             clock_drift_processor.process_timestamp(&mut bucket.timestamp);
             if !keep_metadata {
-                bucket.metadata = BucketMetadata::new();
+                bucket.metadata = BucketMetadata::default();
```

Comment:
`keep_metadata` is already the signal you need instead of `override_received_at_metadata`.